src/stcal/resample/resample.py

```diff
@@ -0,0 +1,1530 @@
+                    "in a data model."
+                )
+
+    def add_model(self, model):
+        """ Resamples model image and either variance data (if ``enable_var``
+        was `True`) or error data (if ``enable_err`` was `True`) and adds
+        them using appropriate weighting to the corresponding
+        arrays of the output model. It also updates resampled data weight,
+        the context array (if ``enable_ctx`` is `True`), relevant output
+        model's values such as "n_coadds".
+
+        Whenever ``model`` has a unique group ID that was never processed
+        before, the "pointings" value of the output model is incremented and
+        the "group_id" attribute is updated. Also, time counters are updated
+        with new values from the input ``model`` by calling
+        :py:meth:`~Resample.update_time`.
+
+        Parameters
+        ----------
+        model : dict
+            A dictionary containing data arrays and other meta attributes
+            and values of actual models used by pipelines.
+
+        """
+        if self._finalized:
+            raise RuntimeError(
+                "Resampling has been finalized and intermediate arrays have "
+                "been freed. Unable to add new models. Call 'reset_arrays' "
+                "to initialize a new output model and associated arrays."
+            )
+        self.validate_input_model(model)
+        self._n_res_models += 1
+
+        data = model["data"]
+        wcs = model["wcs"]
+
+        # Check that input models are 2D images
+        if data.ndim != 2:
+            raise RuntimeError(
+                f"Input model '{model['filename']}' is not a 2D image."
+            )
+
+        self._output_model["n_coadds"] += 1
+
+        if (group_id := model["group_id"]) not in self._group_ids:
+            self.update_time(model)
+            self._group_ids.append(group_id)
+            self.output_model["pointings"] += 1
+
+        iscale = self._get_intensity_scale(model)
+        log.debug(f'Using intensity scale iscale={iscale}')
+
+        pixmap = calc_pixmap(
+            wcs,
+            self.output_model["wcs"],
+            data.shape,
+        )
+
+        log.info("Resampling science and variance data")
+
+        weight = self.build_driz_weight(
+            model,
+            weight_type=self.weight_type,
+            good_bits=self.good_bits,
+        )
+
+        # apply sky subtraction
+        blevel = model["level"]
+        if not model["subtracted"] and blevel is not None:
+            data = data - blevel
+            # self._output_model["subtracted"] = True
+
+        xmin, xmax, ymin, ymax = resample_range(
+            data.shape,
+            wcs.bounding_box
+        )
+
+        add_image_kwargs = {
+            'exptime': model["exposure_time"],
+            'pixmap': pixmap,
+            'scale': iscale,
+            'weight_map': weight,
+            'wht_scale': 1.0,  # hard-coded for JWST count-rate data
```

Comment:
Is this okay for Romancal?  If not, perhaps now is the time to allow it as a variable

Reply:
seems to be OK: https://github.com/spacetelescope/romancal/blob/e250097fb7113be808836a796dc73ba7f687c7b8/romancal/resample/gwcs_drizzle.py#L206

Reply:
great, thanks for checking. maybe update the comment to say that it was also tested for Roman, otherwise it looks like it may have been accidentally ported over from jwst without additional checking

Reply:
> maybe update the comment to say that it was also tested for Roman, otherwise it looks like it may have been accidentally ported over from jwst without additional checking

I did not test it and I am not aware of this being tested but _I assume_ the resample code was tested as a whole by Roman. I actually do not see under what circumstances ``wht_scale`` would need to be `!= 1`. I think the comment should be removed as I think units of the data are irrelevant here.